bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/catalog/Catalog.scala

```diff
@@ -1,80 +1,45 @@
 package edu.gemini.pit.catalog
 
-import scala.actors.Actor._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+import scalaz._
+import Scalaz._
 
 object Catalog {
 
   // An empty catalog that always returns an error
   def empty(msg: String) = new Catalog {
-    def find(id: String)(callback: Callback) {
-      actor {
-        callback.safe(Error(new RuntimeException(msg)))
-      }
-    }
-    override def ||(outer: Catalog): Catalog = outer
-    override def &&(outer: Catalog): Catalog = outer
+
+    override def find(id: String)(implicit ex: ExecutionContext): Future[Result] =
+      Future.successful(Error(new RuntimeException(msg)))
+
+    override def ||(outer: Catalog): Catalog =
+      outer
+
   }
 
 }
 
 trait Catalog { inner =>
 
   /** Given an id (a search string), do a lookup and invoke the callback at some later time. */
-  def find(id: String)(callback: Callback)
+  def find(id: String)(implicit ex: ExecutionContext): Future[Result]
 
   /** Return a new catalog that returns the first result from LHS *or* RHS. */
   def ||(outer: Catalog): Catalog = new Catalog {
 
-    def find(id: String)(callback: Callback) {
-      inner.find(id) { case m => accum ! m }
-      outer.find(id) { case m => accum ! m }
-      lazy val accum = actor {
-        react {
-          case s: Success => callback.safe(s)
-          case _ => react {
-            case r: Result => callback.safe(r)
-          }
-        }
-      }
-    }
-  }
-
-  /** Return a new catalog that returns the merged results from both LHS *and* RHS. */
-  def &&(outer: Catalog): Catalog = new Catalog {
+    def find(id: String)(implicit ex: ExecutionContext): Future[Result] = {
+      val fs = List(inner, outer).map(_.find(id))
 
-    def find(id: String)(callback: Callback) {
-      inner.find(id) { case (crA) => accum ! (('a, crA)) }
-      outer.find(id) { case (crB) => accum ! (('b, crB)) }
-      lazy val accum = actor {
-        await {
-          case ('a, a: Result) => await {
-            case ('b, b: Result) =>
-              callback.safe(merge(a, b))
-              exit()
-          }
-        }
+      Future.find(fs) {
```

Comment:
:+1: I had not noticed this combinator before. I guess we do very little with `Future`.